### ISSUE TITLE
Using `route :any` causes OPTIONS and HEAD requests to fail

### ISSUE THREAD

**Issue by ekampp** [ruby-grape/grape#1056]
According to the documentation, any GET endpoint also serves OPTIONS and HEAD requests. 

I would expect these to be hit before the `route :any` at the bottom of my api. But this isn't the case.

Is this a bug or intentional? If it's intentional, what's the right way to catch stray requests without overriding OPTIONS and HEAD for the real endpoints? 

```ruby
route :any, '*path' do
  raise NotFound, 'Path not NotFound'
end
```

[Here is a demo api](https://github.com/ekampp/grape-options-demo). I would expect this request to return the `Allow` header for the `/names` endpoint. But instead it return `I'm not here`, which means it hits the `route :any` fallback.

```http
OPTIONS /names
```

**Comment by dblock** [comment]
You should turn it into a test, do GETs behave differently from HEADs?

**Comment by ekampp** [comment]
@dblock any suggestions as to where to start adding a test? 

If I understand your question right, then yes: HEAD behaves differently from GET. HEAD doesn't return any response body - only headers. It's so to speak a compacted request. Useful for checking if a cache is busted, and a new one should be fetched.

**Comment by dblock** [comment]
For now just add a standalone test in https://github.com/intridea/grape/tree/master/spec/grape/integration.

**Comment by arempe93** [comment]
@ekampp @dblock closed by #1263?

**Comment by dblock** [comment]
Yes.
